cmd/filestransfer/filestransfer.go

```diff
@@ -0,0 +1,180 @@
+		log.Fatalf("error during writing data to temp file; err: %v", err)
+	}
+	absolutePath := tempFile.Name()
+	fmt.Printf("Tmp file with data to upload created at %v with content %#v\n", absolutePath, customFormatCsvDataToUpload)
+	return absolutePath
+}
+
+func decompressAndRead(file *os.File) (string, error) {
+	gzipReader, err := gzip.NewReader(file)
+	defer gzipReader.Close()
+	if err != nil {
+		return "", err
+	}
+	var b bytes.Buffer
+	_, err = b.ReadFrom(gzipReader)
+	if err != nil {
+		return "", err
+	}
+	return b.String(), nil
+}
+
+func printRows(rows *sql.Rows) {
+	for i := 1; rows.Next(); i++ {
+		var col1, col2 string
+		if err := rows.Scan(&col1, &col2); err != nil {
+			log.Fatalf("error while scaning rows; err: %v", err)
+		}
+		fmt.Printf("Row %v: %v, %v\n", i, col1, col2)
+	}
+}
+
+func main() {
+	if !flag.Parsed() {
+		flag.Parse()
+	}
+
+	//Opening connection
+	dsn, cfg, err := getDSN()
+	if err != nil {
+		log.Fatalf("error while creating DSN from config: %v, error: %v", cfg, err)
+	}
+	db, err := sql.Open("snowflake", dsn)
+	defer db.Close()
+
+	//Creating table to which the data from CSV file will be copied
+	_, err = db.Exec("CREATE OR REPLACE TABLE GOSNOWFLAKE_FILES_TRANSFER_EXAMPLE(num integer, text varchar);")
+	if err != nil {
+		log.Fatalf("error while creating table; err: %v", err)
+	}
+	defer db.Exec("DROP TABLE IF EXISTS GOSNOWFLAKE_FILES_TRANSFER_EXAMPLE;")
+
+	//Uploading data_to_upload.csv to internal stage for table GOSNOWFLAKE_FILES_TRANSFER_EXAMPLE
+	tmpFilePath := tmpFileWithDataToUpload()
+	defer os.Remove(tmpFilePath)
+	_, err = db.Exec(fmt.Sprintf("PUT file://%v @%%GOSNOWFLAKE_FILES_TRANSFER_EXAMPLE;", tmpFilePath))
+	if err != nil {
+		log.Fatalf("error while uploading file; err: %v", err)
+	}
+	fmt.Println("data_do_upload.csv successfully uploaded to internal stage.")
+
+	//Creating custom file format that describes our data
+	_, err = db.Exec("CREATE OR REPLACE TEMPORARY FILE FORMAT CUSTOM_CSV_FORMAT" +
+		" TYPE = CSV COMPRESSION = GZIP FIELD_DELIMITER = ';' FILE_EXTENSION = 'csv' SKIP_HEADER = 1;")
+	if err != nil {
+		log.Fatalf("error while creating file format; err: %v", err)
+	}
+	fmt.Println("Custom CSV format successfully created.")
+
+	//Loading data from files in stage area into table
+	_, err = db.Exec("COPY INTO GOSNOWFLAKE_FILES_TRANSFER_EXAMPLE FILE_FORMAT = CUSTOM_CSV_FORMAT;")
+	if err != nil {
+		log.Fatalf("error while copying data into table; err: %v", err)
+	}
+	fmt.Println("Data successfully loaded into table. Querying...")
+
+	//Querying loaded data from table
+	rows, err := db.Query("SELECT * FROM GOSNOWFLAKE_FILES_TRANSFER_EXAMPLE;")
+	if err != nil {
+		log.Fatalf("error while querying data from table; err: %v", err)
+	}
+	defer rows.Close()
+	printRows(rows)
+
+	//Downloading file from stage area to system's TMP directory
+	tmpDir := os.TempDir()
+	_, err = db.Exec(fmt.Sprintf("GET @%%GOSNOWFLAKE_FILES_TRANSFER_EXAMPLE/data_to_upload.csv file://%v/;", tmpDir))
```

Comment:
I presume we should defer deleting this file?

Reply:
it is bound to internal stage of the table - when we drop table we also drop its stage with files.

Reply:
I meant after downloading :) But it's connected to the discussion above - it's only a temp file, it's not critical.